pkg/storage/bloom/v1/bloom_tokenizer.go

```diff
@@ -68,27 +68,32 @@ func clearCache(cache map[string]interface{}) {
 	}
 }
 
-func calculatePrefix(chk logproto.ChunkRef) []byte {
-	i64buf := make([]byte, binary.MaxVarintLen64)
-	i32buf := make([]byte, 4)
-	prefix := make([]byte, 32)
-
-	binary.PutVarint(i64buf, int64(chk.From))
-	prefix = append(prefix, i64buf...)
-	binary.PutVarint(i64buf, int64(chk.Through))
-	prefix = append(prefix, i64buf...)
-	binary.LittleEndian.PutUint32(i32buf, chk.Checksum)
-	prefix = append(prefix, i32buf...)
-
-	return prefix
```

Comment:
This does a few funny things like using intermediate buffers for int encoding, but mostly it returns the prefix after only allocating 32 bytes which is only enough for the chunk information (ignoring varint-encoding, which makes this more complex). Instead, we want to provision enough space for both the prefix and the maximum expected token length. Otherwise, each iteration of the token iterator can and likely will need to reallocate a new slice of larger length, especially when the initial one has a power-of-2 capacity, which we do in this case (32).